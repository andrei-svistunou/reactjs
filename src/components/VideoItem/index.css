li {
  list-style-type: none;
}

.video__item {
  display: block;
  width: 33.33%;
  padding: 20px;
  cursor: pointer;
}

.video__item a {
  display: block;
}

.video-info.flex {
  align-items: flex-start;
  padding-top: 7px;
}

.video-title,
.video-release {
  font-size: 14px;
}

.video-title {
  padding-top: 2px;
  padding-bottom: 7px;
}

.video-genre {
  font-size: 12px;
}

.video-release {
  border: 1px solid #000;
  border-radius: 2px;
  padding: 2px 5px;
}

.video-img {
  width: 100%;
}