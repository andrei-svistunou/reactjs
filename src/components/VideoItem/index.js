import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './index.css';

const VideoItem = props => (
  <li className='video__item' onClick={props.chooseFilm}>
    <Link to={`/film/${props.id}`}>
      <img className="video-img" src={props.poster}/>
      <div className="video-info flex">
        <div>
          <h4 className="video-title">{props.title}</h4>
          <p className="video-genre">{props.genre}</p>
        </div>
        <span className="video-release">{props.releaseDate}</span>
      </div>
    </Link>
  </li>
);

VideoItem.propTypes = {
  title: PropTypes.string.isRequired,
  releaseDate: PropTypes.string.isRequired,
  genre: PropTypes.string.isRequired,
  poster: PropTypes.string.isRequired,
  chooseFilm: PropTypes.func.isRequired,
  id: PropTypes.number.isRequired
};

export default VideoItem;
