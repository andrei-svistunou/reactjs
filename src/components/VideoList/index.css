.video__list.flex {
  flex-wrap: wrap;
  align-items: flex-start;
}