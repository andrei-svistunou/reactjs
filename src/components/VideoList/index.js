import React from 'react';
import PropTypes from 'prop-types';
import VideoItem from '../VideoItem';
import './index.css';

const VideoList = (props) => {
  const { videoList, chooseFilm } = props;

  return (
    <ul className="video__list flex">
      {
        videoList.map(item =>
        <VideoItem
          key={item.id}
          id={item.id}
          title={item.title}
          genre={item.genres.join('&')}
          poster={item.poster_path}
          releaseDate={item.release_date.slice(0, 4)}
          chooseFilm={() => chooseFilm(item.id)}
        />)
      }
    </ul>
  );
};

VideoList.propTypes = {
  videoList: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    release_date: PropTypes.string.isRequired,
    genres: PropTypes.array.isRequired,
    poster_path: PropTypes.string.isRequired
  })),
  chooseFilm: PropTypes.func.isRequired
};

export default VideoList;
