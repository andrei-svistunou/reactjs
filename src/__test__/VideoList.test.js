import React from 'react';
import { mount } from 'enzyme';
import VideoList from '../components/VideoList';
import VideoItem from '../components/VideoItem';

describe('<VideoList />', () => {
  const Props = {
    videoList: [{
      title: 'Video list',
      release_date: '1990-23-02',
      genres: ['Drama', 'Adventure'],
      poster_path: 'https://path.com/poster/2',
      id: 3
    }],
    chooseFilm: jest.fn()
  };

  test('should render <VideoList />', () => {
    const videoList = mount(<VideoList {...Props}/>);

    expect(videoList.contains(VideoItem)).toBe(true);
    expect(videoList).toMatchSnapshot();
  });

  it('should handle a child click', () => {
    const wrapper = mount(<VideoList {...Props} />);
    wrapper.find(VideoItem).simulate('click');
    expect(Props.chooseFilm).toHaveBeenCalled();
  });
});

