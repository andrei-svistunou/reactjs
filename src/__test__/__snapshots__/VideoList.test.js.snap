// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoList /> should render <VideoList /> without children 1`] = `
<VideoList
  chooseFilm={[MockFunction]}
  videoList={
    Array [
      Object {
        "genres": Array [
          "Drama",
          "Adventure",
        ],
        "id": 3,
        "poster_path": "https://path.com/poster/2",
        "release_date": "1990-23-02",
        "title": "Video list",
      },
    ]
  }
>
  <ul
    className="video__list flex"
  >
    <VideoItem
      chooseFilm={[Function]}
      genre="Drama&Adventure"
      id={3}
      key="3"
      poster="https://path.com/poster/2"
      releaseDate="1990"
      title="Video list"
    >
      <li
        className="video__item"
        onClick={[Function]}
      >
        <img
          className="video-img"
          src="https://path.com/poster/2"
        />
        <div
          className="video-info flex"
        >
          <div>
            <h4
              className="video-title"
            >
              Video list
            </h4>
            <p
              className="video-genre"
            >
              Drama&Adventure
            </p>
          </div>
          <span
            className="video-release"
          >
            1990
          </span>
        </div>
      </li>
    </VideoItem>
  </ul>
</VideoList>
`;
